zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/store/ElasticsearchBatchRequest.java

```diff
@@ -188,14 +191,29 @@ private void validateNoErrors(
       return;
     }
 
-    final String errorMessages =
+    final Map<ErrorKey, List<String>> groupedErrors =
         errorItems.stream()
+            .collect(
+                Collectors.groupingBy(
+                    item -> new ErrorKey(item.operationType(), item.index(), item.error().reason()),
+                    LinkedHashMap::new,
+                    Collectors.mapping(BulkResponseItem::id, Collectors.toList())));
+
+    final String errorMessages =
+        groupedErrors.entrySet().stream()
+            // these error groups can sometimes be too large and even
+            // overflow the logger buffer.
+            .limit(100)
             .map(
-                item ->
+                entry ->
                     String.format(
-                        "%s failed for type [%s] and id [%s]: %s",
-                        item.operationType(), item.index(), item.id(), item.error().reason()))
+                        "%s failed for type %s with ids: %s: %s",
```

Comment:
I know this is not new, but I think we should change it to this:
```suggestion
                        "%s failed on index %s with ids: %s: %s",
```